code/modules/chemistry/Chemistry-Recipes.dm

```diff
@@ -2850,6 +2850,29 @@ datum
 			result_amount = 3
 			mix_phrase = "A white crystalline substance condenses out of the mixture."
 			mix_sound = 'sound/misc/fuse.ogg'
+		
+		slow_saltpetre 
+			name = "slow saltpetre"
+			id = "slow_saltpetre"
+			result = "saltpetre" 
+			// fungus turns compost into ammonium
+			// compost bacteria turns ammonium into nitrates
+			// nitrates are extracted from "soil" with water
+			// potash purifies nitrates into saltpetre
+			required_reagents = list("space_fungus" = 1, "poo" = 1, "potash" = 1, "water" = 1)
+			result_amount = 2
+			instant = 0 // Potash filtering takes time.
+			reaction_speed = 1
+			mix_phrase = "A putrid odor pours from the mixture as a white crystalline substance leaches into the water."
+			mix_sound = 'sound/misc/fuse.ogg'
+			
+			on_reaction(var/datum/reagents/holder, created_volume)
+				// water byproduct
+				// some nitrification processes create additional water.
+				holder.add_reagent("water", created_volume,,holder.total_temperature)
+				// disgusting
+				var/turf/location = get_turf(holder.my_atom)
+				location.fluid_react_single("miasma", created_volume, airborne = 1)
```

Comment:
this won't work when in fluids. consider `pick(holder.covered_turf())` instead

Reply:
Should be fixed now, thank you for the feedback.